internal/pkg/agent/application/monitoring/liveness.go

```diff
@@ -79,6 +83,11 @@ func livenessHandler(coord CoordinatorState) func(http.ResponseWriter, *http.Req
 				unhealthyComponent = true
 			}
 		}
+		if state.OTelStatus != nil {
```

Comment:
I've seen this `OtelStatus != nil` twice and I worry it might be something we can forget to check easily. Would be simpler to have a no-op status reporter when there isn't an otel collector component to eliminate the need for this check everywhere?

Reply:
Also, perhaps worth an IsFailed or IsDegraded otelhelper to simplify the mapping of otel errors to agent states.

Reply:
Please elaborate, I do not understand the ask here.

Reply:
Nevermind, I was misreading this. My head was still a few files earlier.